behavior.py

```diff
@@ -0,0 +1,103 @@
+import datajoint as dj
+
+import actions
+import reference
+import equipment
+
+
+schema = dj.schema(dj.config['names.%s' % __name__], locals())
+
+'''
+TODO: TimeScale items missing wrt TimeSeries items, which were:
+- PupilTracking.xyd
+- PupilTracking.movie
+- HeadTracking.xy_theta
+- HeadTracking.movie
+TimeScale not yet defined
+'''
+
+
+@schema
+class Movie(dj.Manual):
+    # created to track movies
+    definition = """
+    movie_id:                   int             # movie id
+    ---
+    movie_data:                 longblob        # movie data
+    """
+
+
+@schema
+class PupilTracking(dj.Manual):
+    # <class 'behavior.models.PupilTracking'>
+    definition = """
+    -> actions.Session
+    start_time:                 datetime        # start time
+    eye:                        enum("L", "R")  # eye
+    ---
+    -> Movie
+    x_y_d:			longblob        # x y data
+    """
+
+
+@schema
+class HeadTracking(dj.Manual):
+    # <class 'behavior.models.HeadTracking'>
+    definition = """
+    -> actions.Session
+    start_time:                 datetime        # start time
+    ---
+    -> Movie
+    x_y_theta:			longblob        # x y theta
+    """
+
+
+@schema
+class OptogeneticStimulus(dj.Manual):
+    # <class 'behavior.models.OptogeneticStimulus'>
+    definition = """
+    -> actions.Session
+    -> reference.BrainLocation
+    ---
+    -> equipment.Appliance
+    light_delivery:		varchar(255)	# light delivery
+    wavelength:    		float		# wavelength
+    power_calculation_method:	varchar(255)	# power calculation method
+    description:		varchar(255)	# description
+    """
+
+    class Time(dj.Part):
```

Comment:
iirc these had a 1:M relationship - e.g. series of stimulation events.. but then why are they blobs. 
Hmm. potentially could be combined into 1 part table either way.. will audit w/r/t source

Reply:
Yes, if they are 1:M relationship then this could make more sense. But even then, it's weird to me that they are separate 1:M - if anything it sounds like there should be single `Part` table (say `Pulse`) that then has an additional primary key attribute (to make for valid 1:M) and three dependent fields: time, position and power. Let's discuss this point.

Reply:
xcheck w/r/t alyx has:
```
    stimulus_times = models.ForeignKey(Dataset, blank=True, null=True,
                                       related_name="optogenetic_stimulus_times",
                                       help_text="link to an n*2 array of start and stop "
                                       "of each pulse (sec)")
    stimulus_positions = models.ForeignKey(Dataset, blank=True, null=True,
                                           related_name="optogenetic_stimulus_positions",
                                           help_text="link to an n*3 array of stimulus positions")
    power = models.ForeignKey(Dataset, blank=True, null=True,
                              related_name="optogenetic_stimulus_power",
                              help_text="link to an n*1 array giving each pulse power")
```
so, will combine as ```Pulse``` per suggestion since this is in line with your thinking